wireapp/wire-android
# fix: not able to open some conversations [WPB-11325]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11325" title="WPB-11325" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-11325</a>  [Android] Can not open conversations
  </td></table>
  <br />
 




----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

When there are 2 logged in accounts and user receives a full screen intent call for not currently selected account, then after that it's only possible to open conversations where both accounts are members of.

### Causes (Optional)

When a CallActivity is created, it checks whether it's needed to switch account (if the call is for other account), so when user receives a call from a conversation where both accounts are in, it's 50% chance that full screen intent will appear for one or another account (depends which incoming call gets handled first). CallActivity lives in another thread, so when the account is switched and there is another WireActivity in the background, state of both is unsynced and WireActivity still holds previous account from before switching.

### Solutions

Implement observer to keep both activities in sync, so when CallActivity switches account it informs WireActivity and it performs navigation to clear the back stack and recompose the home screen with switched account.
Also, functions to observe the screenshot censoring and sync state is fixed to observe account changes, previously it was taking only single first current account value so it wasn't working when account is switched.
Show incoming call as a full screen intent only for calls for current session, as if shown for another session it will switch to that session and the user won't know that he/she is receiving a call as a different account. For calls that are not for the current session it will show the notification as a heads up notification.
We should also consider showing current account name and/or avatar on CallActivity or even make it possible for the user to switch account on incoming call screen if the call can be answered by more than one account. 

### Testing

#### Test Coverage (Optional)


#### How to Test

Have 2 logged in accounts, open the app, turn off the screen, receive a call for the second account and open the app again - it should be switched to that second account.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
